webapp/packages/plugin-resource-manager/src/ScriptsManager/ScriptsManagerService.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { NavTreeResource, NavNodeInfoResource, NavigationTabsService } from '@cloudbeaver/core-app';
+import { injectable } from '@cloudbeaver/core-di';
+import { NotificationService } from '@cloudbeaver/core-events';
+import { SqlEditorNavigatorService, SqlEditorTabService } from '@cloudbeaver/plugin-sql-editor-navigation-tab';
+
+import { ProjectsResource } from '../ProjectsResource';
+import { ResourceManagerResource } from '../ResourceManagerResource';
+import { RESOURCES_NODE_PATH } from '../RESOURCES_NODE_PATH';
+
+const SCRIPT_EXTENSION = '.sql';
+
+@injectable()
+export class ScriptsManagerService {
+  constructor(
+    private readonly navTreeResource: NavTreeResource,
+    private readonly sqlEditorNavigatorService: SqlEditorNavigatorService,
+    private readonly projectsResource: ProjectsResource,
+    private readonly navNodeInfoResource: NavNodeInfoResource,
+    private readonly sqlEditorTabService: SqlEditorTabService,
+    private readonly navigationTabsService: NavigationTabsService,
+    private readonly notificationService: NotificationService,
+    private readonly resourceManagerResource: ResourceManagerResource
+  ) {
+
+  }
+
+  isScript(nodeId: string) {
+    return nodeId.includes(SCRIPT_EXTENSION);
+  }
+
+  async openScript(nodeId: string) {
+    try {
+      const existingTab = this.sqlEditorTabService.sqlEditorTabs.find(
+        tab => tab.handlerState.associatedScriptId === nodeId
+      );
+
+      if (existingTab) {
+        this.navigationTabsService.selectTab(existingTab.id);
+      } else {
+        const scriptValue = await this.readScript(nodeId);
+
+        const node = await this.navNodeInfoResource.load(nodeId);
+        await this.sqlEditorNavigatorService.openNewEditor({
+          name: node.name ?? 'Unknown script',
+          query: scriptValue,
+          associatedScriptId: nodeId,
+        });
+      }
+    } catch (exception) {
+      this.notificationService.logException(exception as any, 'plugin_resource_manager_open_script_error');
+    }
+  }
```

Comment:
please move logic related to tabs to `@cloudbeaver/plugin-sql-editor-navigation-tab-resource`